fragmenter/fragment.py

```diff
@@ -822,6 +832,48 @@ def fragment(self, molecule: Molecule) -> FragmentationResult:
 
         raise NotImplementedError()
 
+    def fragment(
+        self,
+        molecule: Molecule,
+        toolkit_registry: Optional[Union[ToolkitRegistry, ToolkitWrapper]] = None,
+    ) -> FragmentationResult:
+        """Fragments a molecule according to this class' settings.
+
+        Notes
+        -----
+        * This method is currently *not* guaranteed to be thread safe as it uses and
+          modifies the OpenFF toolkits' ``GLOBAL_TOOLKIT_REGISTRY``.
+
+        Parameters
+        ----------
+        molecule
+            The molecule to fragment.
+        toolkit_registry
+            The underlying cheminformatics toolkits to use for things like conformer
+            generation, WBO computation etc. If no value is provided, the current
+            ``GLOBAL_TOOLKIT_REGISTRY`` will be used. See the OpenFF toolkit
+            documentation for more information.
+
+        Returns
+        -------
+            The results of the fragmentation including the fragments and provenance
+            about the fragmentation.
+        """
+
+        if toolkit_registry is None:
+            toolkit_registry = GLOBAL_TOOLKIT_REGISTRY
+
+        with global_toolkit_registry(toolkit_registry):
+
+            result = self._fragment(molecule)
+
+            result.provenance["toolkits"] = [
+                (toolkit.__class__.__name__, toolkit.toolkit_version)
+                for toolkit in GLOBAL_TOOLKIT_REGISTRY.registered_toolkits
+            ]
```

Comment:
If `toolkit_registry=None`, this will store all toolkits, not just the ones that are used?